ONSdigital/dp-api-tests
# Feature/end to end acceptance tests

### What

Added sunny day scenario of the backend and api's as an end to end test to capture the import and export process (hierarchies and search not implemented yet and will be added as separate pieces of work).

Other changes include updating some of the other API tests to use configured mongo databases.

### How to review

Run tests and check they pass:

1) Make sure you have the following environment variables set:
```
export MONGODB_DATABASE=test
export MONGODB_FILTERS_DATABASE=test
export MONGODB_IMPORTS_DATABASE=test
```
Please read README.md for services to be run

2) Checkout the following branches on the following repositories:
* dp-dataset-api - `feature/generate-downloads`
* dp-filter-api - `feature/generate-downloads`
* dp-dataset-exporter-xlsx - `feature/latest-dp-filter`
* dp-dataset-exporter - `feature/generate-downloads`
* All other services should run on cmd-develop but will need to access the environment variables stated above.

3) Run the following test suits and make sure they pass:
* cd filterAPI; go test ./...
* cd datasetAPI; go test ./...
* cd generateFiles; go test ./...

4) Check mongo has not left any documents in test database and not polluted other databases used for manual testing

5) Unset your environment variables:
```
unset MONGODB_DATABASE
unset MONGODB_FILTERS_DATABASE
unset MONGODB_IMPORTS_DATABASE
```

### Who can review

Team B
